pvlib/iotools/pvgis.py

```diff
@@ -23,6 +23,353 @@
+        Orientation (azimuth angle) of the (fixed) plane. 0=south, 90=west,
+        -90: east. Not relevant for tracking systems.
+    outputformat: str, default: 'json'
+        Must be in ``['json', 'csv']``. See PVGIS hourly data
+        documentation [2]_ for more info.
+    usehorizon: bool, default: True
+        Include effects of horizon
+    userhorizon: list of float, default: None
+        Optional user specified elevation of horizon in degrees, at equally
+        spaced azimuth clockwise from north, only valid if `usehorizon` is
+        true, if `usehorizon` is true but `userhorizon` is `None` then PVGIS
+        will calculate the horizon [4]_
+    raddatabase: str, default: None
+        Name of radiation database. Options depend on location, see [3]_.
+    start: int, default: None
+        First year of the radiation time series. Defaults to first year
+        avaiable.
+    end: int, default: None
+        Last year of the radiation time series. Defaults to last year avaiable.
+    pvcalculation: bool, default: False
+        Return estimate of hourly PV production.
+    peakpower: float, default: None
+        Nominal power of PV system in kW. Required if pvcalculation=True.
+    pvtechchoice: {'crystSi', 'CIS', 'CdTe', 'Unknown'}, default: 'crystSi'
+        PV technology.
+    mountingplace: {'free', 'building'}, default: free
+        Type of mounting for PV system. Options of 'free' for free-standing
+        and 'building' for building-integrated.
+    loss: float, default: None
+        Sum of PV system losses in percent. Required if pvcalculation=True
+    trackingtype: {0, 1, 2, 3, 4, 5}, default: 0
+        Type of suntracking. 0=fixed, 1=single horizontal axis aligned
+        north-south, 2=two-axis tracking, 3=vertical axis tracking, 4=single
+        horizontal axis aligned east-west, 5=single inclined axis aligned
+        north-south.
+    optimal_surface_tilt: bool, default: False
+        Calculate the optimum tilt angle. Not relevant for 2-axis tracking
+    optimalangles: bool, default: False
+        Calculate the optimum tilt and azimuth angles. Not relevant for 2-axis
+        tracking.
+    components: bool, default: True
+        Output solar radiation components (beam, diffuse, and reflected).
+        Otherwise only global irradiance is returned.
+    url: str, default:const:`pvlib.iotools.pvgis.URL`
+        Base url of PVGIS API, append ``seriescalc`` to get hourly data
+        endpoint
+    map_variables: bool, default True
+        When true, renames columns of the Dataframe to pvlib variable names
+        where applicable. See variable VARIABLE_MAP.
+    timeout: int, default: 30
+        Time in seconds to wait for server response before timeout
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        Time-series of hourly data, see Notes for fields
+    inputs : dict
+        Dictionary of the request input parameters
+    metadata : list or dict
```

Comment:
I think this is only ever a dict, when can it be a list?

Reply:
Nope, you are right. It was an artefact copied from get_pvgis_tmy where the csv option is returning metadata as a list.